sky/shell/platform/ios/accessibility_bridge.h

```diff
@@ -0,0 +1,177 @@
+  FlutterView* view_;
+  NodeMap nodes_;
+
+  mojo::StrongBinding<semantics::SemanticsListener> binding_;
+
+  base::WeakPtrFactory<AccessibilityBridge> weak_factory_;
+
+  DISALLOW_COPY_AND_ASSIGN(AccessibilityBridge);
+};
+
+struct Flags {
+  Flags()
+      : can_be_tapped(false),
+        can_be_long_pressed(false),
+        can_be_scrolled_horizontally(false),
+        can_be_scrolled_vertically(false),
+        has_checked_state(false),
+        is_checked(false) {}
+
+  // Assigns the specified flags to this struct iff it is non-null.
+  // It is important not to update our flags when the semantic flags
+  // are null, because a null update from Dart means "nothing to update"
+  Flags& operator=(const semantics::SemanticFlagsPtr& other) {
+    if (!other.is_null()) {
+      can_be_tapped = other->canBeTapped;
+      can_be_long_pressed = other->canBeLongPressed;
+      can_be_scrolled_horizontally = other->canBeScrolledHorizontally;
+      can_be_scrolled_vertically = other->canBeScrolledVertically;
+      has_checked_state = other->hasCheckedState;
+      is_checked = other->isChecked;
+    }
+    return *this;
+  }
+
+  bool can_be_tapped : 1;
+  bool can_be_long_pressed : 1;
+  bool can_be_scrolled_horizontally : 1;
+  bool can_be_scrolled_vertically : 1;
+  bool has_checked_state : 1;
+  bool is_checked : 1;
+};
+
+struct Strings {
+  // Assigns the specified strings to this struct iff it is non-null
+  // It is important not to update our strings when the semantic strings
+  // are null, because a null update from Dart means "nothing to update"
+  Strings& operator=(const semantics::SemanticStringsPtr& other) {
+    if (!other.is_null()) {
+      if (!other->label.is_null()) {
+        label = other->label.get();
+      }
+    }
+    return *this;
+  }
+
+  std::string label = "";
+};
+
+struct Geometry {
+  // Assigns the specified geometry to this struct iff it is non-null
+  // It is important not to update our values when the semantic geometry
+  // is null, because a null update from Dart means "nothing to update"
+  Geometry& operator=(const semantics::SemanticGeometryPtr& other) {
+    if (!other.is_null()) {
+      if (!other->transform.is_null()) {
+        transform.setColMajorf(other->transform.data());
+      }
+      rect.setXYWH(other->left, other->top, other->width, other->height);
+    }
+    return *this;
+  }
+
+  SkMatrix44 transform = SkMatrix44();
+  SkRect rect = SkRect();
+};
+
+// Class that holds information about accessibility nodes, which are used
+// to construct iOS accessibility elements
+class Node final : public base::RefCounted<Node> {
+ public:
+  static const uint32_t kUninitializedNodeId = -1;
+
+  void Update(const semantics::SemanticsNodePtr& node);
+  void PopulateAccessibleElements(NSMutableArray* accessibleElements);
+
+ private:
+  friend class base::RefCounted<Node>;
+  friend class AccessibilityBridge;
+  typedef std::vector<NodePtr> NodeList;
+
+  Node() = delete;
```

Comment:
There's no need for this statement.  Having a constructor with a non-zero number of argument implicitly deletes the zero-argument constructor.